searchlib/src/tests/docstore/store_by_bucket/store_by_bucket_test.cpp

```diff
@@ -56,33 +57,61 @@ class VerifyBucketOrder : public StoreByBucket::IWrite {
         _lastBucketId = bucketId;
         EXPECT_EQUAL(0, memcmp(data.data(), createPayload(bucketId).c_str(), data.size()));
     }
-    ~VerifyBucketOrder() override;
+
 private:
     uint32_t _lastLid;
     BucketId _lastBucketId;
     vespalib::hash_set<uint32_t> _uniqueUser;
     vespalib::hash_set<uint64_t> _uniqueBucket;
+
 };
 
 VerifyBucketOrder::~VerifyBucketOrder() = default;
 
+struct StoreIndex : public StoreByBucket::StoreIndex {
+    ~StoreIndex() override;
+    void store(const StoreByBucket::Index &index) override {
+        _where.push_back(index);
+    }
+    std::vector<StoreByBucket::Index> _where;
+};
+StoreIndex::~StoreIndex() = default;
+
+struct Iterator : public StoreByBucket::IndexIterator {
+    Iterator(const std::vector<StoreByBucket::Index> & where) : _where(where), _current(0) {}
+
+    bool has_next() noexcept override {
+        return _current < _where.size();
+    }
+
+    StoreByBucket::Index next() noexcept override {
+        return _where[_current++];
+    }
+
+    const std::vector<StoreByBucket::Index> & _where;
+    uint32_t _current;
+};
+
 TEST("require that StoreByBucket gives bucket by bucket and ordered within")
 {
     std::mutex backing_lock;
     vespalib::MemoryDataStore backing(vespalib::alloc::Alloc::alloc(256), &backing_lock);
     vespalib::ThreadStackExecutor executor(8);
-    StoreByBucket sbb(backing, executor, CompressionConfig::LZ4);
+    StoreIndex storeIndex;
+    StoreByBucket sbb(storeIndex, backing, executor, CompressionConfig::LZ4);
     for (size_t i(1); i <=500; i++) {
         add(sbb, i);
     }
     for (size_t i(1000); i > 500; i--) {
         add(sbb, i);
     }
     sbb.close();
-    EXPECT_EQUAL(32u, sbb.getBucketCount());
-    EXPECT_EQUAL(1000u, sbb.getLidCount());
+    std::sort(storeIndex._where.begin(), storeIndex._where.end());
+    //EXPECT_EQUAL(32u, sbb.getBucketCount());
```

Comment:
Why is this commented out?

Reply:
It does not exist on this level anymore.
See new test of BucketIndexStore.